external/anomaly/tests/test_ote_training.py

```diff
@@ -0,0 +1,339 @@
+    print(f"training_parameters_fields: {training_parameters_fields}")
+    make_paths_be_abs(training_parameters_fields, dataset_definitions[ROOT_PATH_KEY])
+    print(f"training_parameters_fields after make_paths_be_abs: {training_parameters_fields}")
+
+    assert set(DATASET_PARAMETERS_FIELDS()) == set(training_parameters_fields.keys()), \
+            f'ERROR: dataset definitions for name={dataset_name} does not contain all required fields'
+    assert all(training_parameters_fields.values()), \
+            f'ERROR: dataset definitions for name={dataset_name} contains empty values for some required fields'
+
+    params = DatasetParameters(**training_parameters_fields)
+    return params
+
+
+def _create_anomaly_classification_dataset_and_labels_schema(dataset_params):
+    logger.debug(f'Path to dataset: {dataset_params.dataset_path}')
+    category_list = [f.path for f in os.scandir(dataset_params.dataset_path) if f.is_dir()]
+    items = []
+    for category in category_list:
+        if "vivo" not in category:
+            logger.debug(f'Creating dataset for {category}')
+            items.extend(OteMvtecDataset(path=category, seed=0).generate())
+    dataset = DatasetEntity(items=items)
+    labels = dataset.get_labels()
+    labels_schema = LabelSchemaEntity.from_labels(labels)
+    return dataset, labels_schema
+
+
+def get_anomaly_test_action_classes() -> List[Type[BaseOTETestAction]]:
+    return [
+        AnomalyDetectionTestTrainingAction,
+        OTETestTrainingEvaluationAction,
+        OTETestExportAction,
+        OTETestExportEvaluationAction,
+        OTETestPotAction,
+        OTETestPotEvaluationAction,
+        OTETestNNCFAction,
+        OTETestNNCFEvaluationAction,
+        OTETestNNCFExportAction,
+        OTETestNNCFExportEvaluationAction,
+    ]
+
+
+class AnomalyDetectionTrainingTestParameters(DefaultOTETestCreationParametersInterface):
+    def test_case_class(self) -> Type[OTETestCaseInterface]:
+        return generate_ote_integration_test_case_class(
+            get_anomaly_test_action_classes()
+        )
+
+    def test_bunches(self) -> List[Dict[str, Any]]:
+        test_bunches = [
+                dict(
+                    model_name=[
+                       'ote_anomaly_classification_padim',
+                       'ote_anomaly_classification_stfpm',
+                    ],
+                    dataset_name='mvtec',
+                    usecase='precommit',
+                ),
+                dict(
+                    model_name=[
+                       'ote_anomaly_classification_padim',
+                       'ote_anomaly_classification_stfpm',
+                    ],
+                    dataset_name='mvtec',
+                    batch_size=KEEP_CONFIG_FIELD_VALUE,
+                    usecase=REALLIFE_USECASE_CONSTANT,
+                ),
+
+        ]
+        return deepcopy(test_bunches)
+    
+    def short_test_parameters_names_for_generating_id(self) -> OrderedDict:
```

Comment:
Do you really need to override this method?
Is it different w.r.t. `DefaultOTETestCreationParametersInterface`?

Reply:
Yes, the configs for Anomaly detection does not contain any iters/epochs. So we don't need parameter
`("num_training_iters", "num_iters"),`
Method must be overwritten.